code/modules/vehicles/trolley.dm

```diff
@@ -46,6 +46,18 @@
 			playsound(src, pick('sound/effects/wounds/crack1.ogg', 'sound/effects/wounds/crack2.ogg'), 40)
 			animate(src, pixel_y = -4, time = 0.5 SECONDS, easing = SINE_EASING)
 			animate(occupants[1], pixel_y = 4, time = 0.5 SECONDS, easing = SINE_EASING)
+	if(amount_of_cargo)
+		handle_transform()
+
+/obj/vehicle/ridden/trolley/proc/handle_transform(var/unloading)
+	var/matrix/new_transform = matrix()
+	if(dir == WEST)
+		new_transform.Scale(-1,1)
+	else if(unloading || dir != WEST)
+		new_transform.Scale(1,1)
```

Comment:
The second if() will always be true, is this intentional?

Reply:
It flips the crate sprite using Scale(-1,1) when the trolley is facing west, otherwise it will be reset to the original sprite.

Reply:
> It flips the crate sprite using Scale(-1,1) when the trolley is facing west, otherwise it will be reset to the original sprite.

You don't need this else case at all, as a blank matrix is the same as scaling it by 1,1.

Reply:
well spotted, thank you